### PR TITLE
Get plotly figure by title

### PR DESCRIPTION
**Pull request by khameeteman**

### CODE REVIEW

File: excel_graph_parser/graph_parser.py [hunk 1]
```diff
@@ -212,6 +212,13 @@ def get_figures_from_excel_file(self) -> list:
 
         return figures
 
+    def get_plotly_figure_by_title(self, title: str) -> go.Figure:
+        figures = self.get_figures_from_excel_file()
```
Comment (kvangiessen-VIKTOR) [inline excel_graph_parser/graph_parser.py]: I was thinking more along the lines of extracting the chart to figure conversion from the  `get_figures_from_excel_file()`.

Currently it iterates through all charts and creates a figure for each one of them, this causes quite some overhead if there are 10 charts, but you only need one.

Comment (kvangiessen-VIKTOR) [reply excel_graph_parser/graph_parser.py]:
```python

def get_named_charts() -> dict[str, Any]:
    charts = {}
    for i, chart in enumerate(self.charts):
        chart_title = ''.join([title_element.t for title_element in chart.title.tx.rich.p[0].r])  if chart.title else  f"Untitled Chart {i}"
        charts[chart_title] = chart
    return charts

def chart_to_figure(chart, title) -> go.Figure | None:
    ...
    return figure

def get_figures_from_excel_file(self) -> list[go.Figure]:
     """Gets figures from the excel file as a list"""
     figures = []
     for chart_title, chart in enumerate(self.get_named_charts().items()):
        if figure := self.chart_to_figure(chart, title):
            figures.append(figure)
     return figures

def get_plotly_figure_by_title(self, title: str) -> go.Figure:
    named_charts = self.get_named_charts()
    chart = named_charts[title]
    return chart_to_figure(chart, title)
```

Comment (kvangiessen-VIKTOR) [reply excel_graph_parser/graph_parser.py]: We could even update the init to get self.charts as a mapping between title and chart

Comment (khameeteman) [reply excel_graph_parser/graph_parser.py]: Updated

**Pull request closed without merging**